brouter-core/src/main/java/btools/router/RoutingEngine.java

```diff
@@ -559,6 +564,126 @@ void buildPointsFromCircle(List<OsmNodeNamed> waypoints, double startAngle, doub
+    listStart.add(wpt1);
+
+    List<OsmNodeNamed> wpliststart = new ArrayList<>();
+    wpliststart.add(wp);
+
+    List<OsmNodeNamed> listSearch = new ArrayList<>();
+
+    for (int a = 45; a < 360; a +=90) {
+      int[] pos = CheapRuler.destination(wp.ilon, wp.ilat, searchRadius * 2, a);
+      OsmNodeNamed onn = new OsmNodeNamed(new OsmNode(pos[0], pos[1]));
+      onn.name = "via" + a;
+
+      MatchedWaypoint wpt = new MatchedWaypoint();
+      wpt.waypoint = onn;
+      wpt.name = onn.name;
+      listStart.add(wpt);
+    }
+
+    for (int a = 0; a < 360; a +=90) {
+      int[] pos = CheapRuler.destination(wp.ilon, wp.ilat, searchRadius, a);
+      OsmNodeNamed onn = new OsmNodeNamed(new OsmNode(pos[0], pos[1]));
+      onn.name = "via" + a;
+      listSearch.add(onn);
+    }
+
+    RoutingEngine re = null;
+    RoutingContext rc = new RoutingContext();
+    rc.localFunction = "dummy";
+
+    re = new RoutingEngine(null, null, segmentDir, wpliststart, rc, BROUTER_ENGINEMODE_ROUNDTRIP);
+    rc.useDynamicDistance = true;
+    re.matchWaypointsToNodes(listStart);
+    re.resetCache(true);
+
+    int numForest = rc.expctxWay.getLookupKey("estimated_forest_class");
+    int numRiver = rc.expctxWay.getLookupKey("estimated_river_class");
+
+    OsmNode start1 = re.nodesCache.getStartNode(listStart.get(0).node1.getIdFromPos());
+
+    double elev = (start1 == null ? 0 : start1.getElev());
+
+    List<AreaInfo> ais = new ArrayList<>();
+
+    listStart.remove(0);
+    for (int a = 0; a < 4; a++) {
+      rc.ai = new AreaInfo(a * 90 +90);
+      rc.ai.elevStart = elev;
+      rc.ai.numForest = numForest;
+      rc.ai.numRiver = numRiver;
+
+      List<OsmNodeNamed> wplist = new ArrayList<>();
+      wplist.add(new OsmNodeNamed(listStart.get(a).waypoint));
+      if (a == 3) wplist.add(new OsmNodeNamed(listStart.get(0).waypoint));
+      else        wplist.add(new OsmNodeNamed(listStart.get(a+1).waypoint));
+
+      rc.ai.polygon = new OsmNogoPolygon(true);
+      rc.ai.polygon.addVertex(wp.ilon, wp.ilat);
+      rc.ai.polygon.addVertex(wplist.get(0).ilon, wplist.get(0).ilat);
+      rc.ai.polygon.addVertex(wplist.get(1).ilon, wplist.get(1).ilat);
+
+      List<OsmNodeNamed> wpsearchlist = new ArrayList<>();
+      wpsearchlist.add(listSearch.get(a));
+      if (a == 3) wpsearchlist.add(listSearch.get(0));
+      else        wpsearchlist.add(listSearch.get(a+1));
+
+      rc.useDynamicDistance = true;
+      re = new RoutingEngine(null, null, segmentDir, wpsearchlist, rc, 0);
+      rc.processUnusedTags = true;
+      rc.turnInstructionMode = 9;
+      re.doRun(0);
+
+      ais.add(rc.ai);
+    }
+
+    logInfo("execution time get area = " + (System.currentTimeMillis() - start) / 1000. + " seconds");
+
+
+    // for (AreaInfo ai: ais) {
+    //  System.out.println("\n" + ai.toString());
+    //}
+
+    switch (preferredRandomType) {
+      case AreaInfo.RESULT_TYPE_ELEV50:
+        ais.sort(Comparator.comparingInt(AreaInfo::getElev50Weight));
```

Comment:
@afischerdev `List.sort` requires Android 7.

Better use `Collections.sort` to be compatible with all Android versions.

Reply:
Ok, I'll change